### DISCUSSION TITLE
Removing contexts from Model

### DISCUSSION THREAD

**Discussion by naturzukunft** [eclipse-rdf4j/rdf4j#4849] [Q&A] [answered]
Hi there, is there a way to removing the context in a model? 
Currently i write it to a string with `Rio.write(model, sw, RDFFormat.TURTLE);` 
And then create a model from that string, but i think there is a better way.
Thanks Fredy

**Comment by naturzukunft** [comment]
That is also possible: 
```
Model copy = new DynamicModelFactory().createEmptyModel();
model.forEach(stmt->copy.add(stmt.getSubject(), stmt.getPredicate(), stmt.getObject()));
```

**Comment by hmottestad** [comment, marked as answer]
We actually have a method for stripping the context from an entire model:  `Models.stripContexts(Model model,
 Resource... contexts)`


<img width="919" alt="Screenshot 2023-11-18 at 19 08 28" src="https://github.com/eclipse-rdf4j/rdf4j/assets/797185/bf34df43-1132-4448-920c-62762b074da6">


https://rdf4j.org/javadoc/latest/org/eclipse/rdf4j/model/util/Models.html#stripContexts(org.eclipse.rdf4j.model.Model,org.eclipse.rdf4j.model.Resource...)

**Reply by hmottestad** [reply]
We also have `Statements.stripContext(stmt)` if you want to strip the context from a single statement.

https://rdf4j.org/javadoc/latest/org/eclipse/rdf4j/model/util/Statements.html#stripContext(org.eclipse.rdf4j.model.Statement)
